src/main/java/games/strategy/engine/framework/headlessGameServer/AvailableGames.java

```diff
@@ -75,11 +80,30 @@ String getGameFilePath(final String gameName) {
   private static void populateAvailableGames(final Map<String, URI> availableGames,
       final Set<String> availableMapFolderOrZipNames, final Set<String> mapNamePropertyList) {
     System.out.println("Parsing all available games (this could take a while). ");
-    for (final File map : allMapFiles()) {
-      if (map.isDirectory()) {
-        populateFromDirectory(map, availableGames, availableMapFolderOrZipNames, mapNamePropertyList);
-      } else if (map.isFile() && map.getName().toLowerCase().endsWith(ZIP_EXTENSION)) {
-        populateFromZip(map, availableGames, availableMapFolderOrZipNames, mapNamePropertyList);
+    final List<File> files = allMapFiles();
+    if (files.size() > 0) {
+      final ExecutorService service = Executors.newWorkStealingPool(files.size());
+      final List<Callable<Object>> tasks = new ArrayList<>(files.size());
```

Comment:
It's a shame there's no `invokeAll()` overload for a collection of `Runnable`s.  Changing the generic type parameter to `Void` might better convey the intention that there is no return value from these tasks.

Reply:
Good suggestion, my first attempt was using <?> as type, but error-prone complained so I needed to change it to object